### ISSUE TITLE
Inconsistent password terminology

### ISSUE THREAD

**Issue by Forage** [m2049r/xmrwallet#284]
- `Wallet Files Restore Password` in de wallet details
- `Passphrase` in de wallet details menu
- `Password` in the dialogue for accessing the wallet

It's all not really helping to make things understandable for people when using different ways to label the same thing.

I suggest sticking to one thing, simply `password` or `wallet password` it is known by all users and less technical (sounding) than `passphrase`. It's this password that allows you to access the wallet. That this means that the wallet files are encrypted and giving the password will allow the app to decrypt them again is not that important from a UI perspective. That opening is basically the same thing as recovering does not need to be emphasised either.

With the introduction of the 'CrAzYpass' feature, things got even more complicated. No matter how clever it might be, in the end it is nothing more than a fancy hashing algorithm of the user's password. It being important or not, a big achievement or not, the user does not need to be reminded of how the hash was generated. Giving it a fancy name will only raise questions and create confusion. Labelling it what it is, a `password hash`, is short, clear and effective.

**Comment by apertamono** [comment]
I was going to ask whether 'passphrase' is the same thing as 'password'. I guess this is the answer.

@Forage Would you like to review the Dutch translation next week?

**Comment by m2049r** [comment]
no, this is not the answer. the "password" is what the user enters to open the wallet. the "passphrase" is what is actually used to encrypt the wallet. the usage is consistent. in order to open a wallet you copy to the pc or a new phone you need the passphrase. not the password.

yes, it is complicated. but it must be clear that the password IS NOT the passphrase. so no, these are not two words to label the same thing.

it is very important to keep this distinction in translations.

**Comment by Forage** [comment]
When you create a wallet you are only asked to provide a wallet passphrase. The same value has to be used later on to open the wallet...where the dialog asks you to enter your password.
There is also no way to change you password in the app, only your passphrase.

In its current situation they are one and the same from the user's point of view. It should therefore be called the same as well, in my opinion.
If they are technically not the same and it is really that important that the user understands this and it should be kept like this than I think this should either be made clear by an explanation or you should provide means to provide/change different values for the two.

**Comment by apertamono** [comment]
@m2049r: 
> no, this is not the answer. the "password" is what the user enters to open the wallet. the "passphrase" is what is actually used to encrypt the wallet.

OK, if you say so. But the help page repeatedly says that it's the user's choice whether to use a password or a passphrase:

`Use a strong password - even better use a passphrase.`

And in other help texts, *password* and *passphrase* are used interchangeably:

```
<h3>Legacy Password</h3>
        If you see your passphrase here, your wallet files are not as secure as when using
        a CrAzYpass.
To fix this, simply select \"Change Passphrase\" from the menu.
```

```
By selecting \"Change Passphrase\" from the menu, you can choose another passphrase.
Beware that this will generate a new Recovery Password.
```

Can you explain this to me as if I was an iPhone user, and list how many types of passwords you're using?

1. 25-word mnemonic seed.
2. 52-character CrAzYpass hash.
3. Password?
4. Passphrase?
5. Recovery password???

**Comment by m2049r** [comment]
@ProkhorZ Good points all. The discrepancies you point out need fixing (they are probably from a time when we had a single not-so-secure password).

We are using:
1. 25-word mnemonic seed
2. 52-character recovery **password** (this is the real wallet password)
3. user **passphrase** (this is what the user needs to remember & type)

(1) & (2) should be written down by the user since (2) can also be used instead of (3) - this is useful if moving your wallet between devices. (1) is sufficient if the user doesn't care for moving wallets around.

I would be happy if you find better nomenclature.

**Comment by m2049r** [comment]
@Forage 

> There is also no way to change you password in the app, only your passphrase.

This is only half-true. Changing the passphrase changes the password.

Yes - it needs more/better explanation. If we could somehow link the "password" to "moving wallets around between devices or recovering backups" ...

Maybe "CrAzYpass" & "passphrase" could be simply used to denote the two? Might work with good & consistent usage in the help files & labels.

**Comment by Forage** [comment]
> > There is also no way to change you password in the app, only your passphrase.
> 
> This is only half-true. Changing the passphrase changes the password.

My point exactly. And since I haven't change my password yet they are one and the same.

> Maybe "CrAzYpass" & "passphrase" could be simply used to denote the two? Might work with good & consistent usage in the help files & labels.

With the introduction of using a password hash to encrypt the wallet things got more complicated. While you partially took over the responsibility of the user, coming up with a strong password, there was not a need to make a distinction before all this.
Moving forward, I'd say:
1. Stick to "password" and "passphrase" now
2. Don't complicate things even more by referring to 'CrAzYpass'
3. Change the menu item for changing the password to "Change password/passphrase"
4. Refer to "password" instead of "passphrase" in the wallet creation dialog
5. Use "Wallet passphrase" in the details dialog with a good explanation in the that the passphrase is a hash of the password, used 'physically' encrypt the wallet
6. Never making the password used to generate the encryption hash visible
7. And, as @ProkhorZ pointed out, be consistent about it in the documentation elsewhere

Or the alternative:
1. Drop the compatibility with the official client in term of a direct import
2. Only refer to "password" everywhere
3. Remove any trace of "passphrase" or it's visibility in the app
4. And not making the password visible in the details dialog again

This way you use the password hash to encrypt the wallet transparently, without bother the user with it.
Yes, the consequence is that the user can no longer directly import the wallet in the official client, but is that such a big deal? Who will actually do this?
You can always provide an export function, simply changing the encryption based on the password again instead of its hash. With this option, in the worst case, one would have to import in the app and export to the official client, just the have access to a backup again. But that's not something you'd do every day/week/month/year(?).

**Comment by apertamono** [comment]
> We are using:
>
> 1. 25-word mnemonic seed
> 2. 52-character recovery password (this is the real wallet password)
> 3. user passphrase (this is what the user needs to remember & type)

Thanks for explaining! My recommendation for the terminology: Call #2 the Crazypass code or passcode. Lay people will call anything that looks like gibberish but has a hidden meaning a code. 

And please don't use 'recovery' for anything except the mnemonic seed, because that's the only piece of data that's enough by itself to recover the funds.

Ideally, the Monero developers would change the term 'mnemonic seed' to 'recovery phrase', like the [new Lightning light wallet](https://blog.lightning.engineering/announcement/2018/09/10/lightning-app.html) did. I already translated 'mnemonic seed' or 'seed' with the equivalent of 'recovery text', because a literal translation wouldn't make sense.

I'm not an expert in security design, but if you think writing down a password is safer than remembering it, maybe don't let users choose their own password at all? Just let them make a gesture or bash some keys for randomness.

**Comment by Mansarde** [comment]
I would rename Crazypass to "Wallet Pass(word)", as that should make it immediately obvious what it is for. It's self-descriptive.

But what to call the user's easier to remember password (which is only valid per Monerujo installation) I'm not too sure about. Maybe "Quick Pass" or "Easy Pass", because it's for quick access without having to enter the actual wallet pass?
The problem with that one is that a short name might not make it immediately obvious that it is only valid until a re-install or when moving the wallet.

I'd definitely call the wallet's actual password "Wallet Pass" though.
That makes it clear what it's for without having to read up on it first.
Or, if you want to encourage users to back it up, call it "Backup Wallet Pass".

**Comment by m2049r** [comment]
Password: what the user enters
Wallet Passcode: the actual wallet password, aka crazypass

would that work? @Mansarde @ProkhorZ @Forage 

if so, i would rewrite the english help files & labels and we take it from there ...
